backend/audit/test_schemas.py

```diff
@@ -403,76 +391,77 @@ def test_all_booleans(self):
-        simple_case = jsoncopy(self.SIMPLE_CASE[1])
-
-        for field in boolean_fields:
-            for value in [True, False]:
-                simple_case[field] = value
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
+                validate(simple_case, schema)
 
     def test_all_gaap_results(self):
+        schema = self.AUDIT_INFO_SCHEMA
         simple_case = jsoncopy(self.SIMPLE_CASE[0])
+        gaap_results = [
+            "unmodified_opinion",
+            "qualified_opinion",
+            "adverse_opinion",
+            "disclaimer_of_opinion",
+        ]
 
-        for _ in range(5):
-            for n in range(1, 5):
-                ls = random.sample(self.ALL_OPINIONS, n)
+        for _ in range(10):
+            for n in range(2, 5):
+                ls = random.sample(gaap_results, n)
                 simple_case["gaap_results"] = ls
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
+                validate(simple_case, schema)
 
-    def test_all_framework_basis(self):
-        simple_case = jsoncopy(self.SIMPLE_CASE[1])
+        # Test when it is 'not_gaap', which requires other fields.
+        simple_case["gaap_results"] = ["not_gaap"]
+        simple_case["is_sp_framework_required"] = random.choice([True, False])
+        simple_case["sp_framework_basis"] = random.choices(
+            ["cash_basis", "tax_basis", "contractual_basis", "other_basis"], k=2
+        )
+        simple_case["sp_framework_opinions"] = random.choices(
+            [
+                "unmodified_opinion",
+                "qualified_opinion",
+                "adverse_opinion",
+                "disclaimer_of_opinion",
+            ],
+            k=2,
+        )
+        validate(simple_case, schema)
 
-        sp_framework_basis = [
-            "cash_basis",
-            "tax_basis",
-            "contractual_basis",
-            "other_basis",
+    def test_bad_gaap_results(self):
+        schema = self.AUDIT_INFO_SCHEMA
+        simple_case = jsoncopy(self.SIMPLE_CASE[0])
+        not_gaap_values = [
+            "state",
+            "local",
+            "tribal",
+            "higher-ed",
+            "non-profit",
+            "unknown",
+            "none",
         ]
 
-        for _ in range(5):
-            for n in range(1, 5):
-                opinions = random.sample(self.ALL_OPINIONS, n)
-                basis = random.sample(sp_framework_basis, n)
-                simple_case["sp_framework_opinions"] = opinions
-                simple_case["sp_framework_basis"] = basis
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
+        for word in not_gaap_values:
+            simple_case["gaap_results"] = [word]
+            self.assertRaises(exceptions.ValidationError, validate, simple_case, schema)
 
-    def test_bad_gaap_results(self):
+    def test_bad_sp_framework_opinions(self):
+        schema = self.AUDIT_INFO_SCHEMA
         simple_case = jsoncopy(self.SIMPLE_CASE[0])
```

Comment:
This should be self.SIMPLE_CASE[1] as `sp_framework_opinions` is only expected when `gaap_results=["not_gaap"]` and that condition is true for self.SIMPLE_CASE[1] but not for self.SIMPLE_CASE[0]. Otherwise a validationError will be triggered as expected but not for the expected reason. FYI, I will make the change